### PR TITLE
Add kitchen tests for agent installation

### PR DESCRIPTION
**Pull request by kbogtob**
### What does this PR do?

It adds kitchen tests and refactor vagrant environments to suit kitchen tests needs. It allows to test only the basic datadog_agent role for now. It adds a specific documentation section about kitchen tests.

### Motivation

In the aim of having uniform tests for our configuration managements, we worked on adding integration tests for this puppet module.

### CODE REVIEW

File: test/integration/dd-agent/serverspec/spec_helper.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+# # :backend can be either :exec or :ssh
+# # since we are running local we use :exec
+set :backend, :exec
+
+# require 'serverspec'
+# require 'pathname'
+# require 'net/ssh'
+
+# RSpec.configure do |config|
```
Comment (albertvaka) [inline test/integration/dd-agent/serverspec/spec_helper.rb]: Is there any reason to leave this commented out code here?

Comment (kbogtob) [reply test/integration/dd-agent/serverspec/spec_helper.rb]: Nope, remains of one of my tries, will clean

File: .gitignore [hunk 2]
```diff
@@ -24,10 +24,11 @@ spec/fixtures
...
-tests/centos/.vagrant
-tests/ubuntu/.vagrant
+environments/**/.vagrant
 
 .DS_Store
 .project
 .envrc
 /inventory.yaml
+
+.kitchen
```
[the remark this answers is not in the document]
Comment (kbogtob) [reply .gitignore]: Good idea, thanks!

File: environments/README.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
-# Test setup
+# [DEPRECATED] Test setup
```
[the remark this answers is not in the document]
Comment (kbogtob) [reply environments/README.md]: Will update

**Comment by albertvaka** [comment]
Can you make the tests part of the CI?

**Comment by kbogtob** [comment]
Sure, I'll create a new PR.

**Pull request merged**